siliconcompiler/tools/openroad/openroad.py

```diff
@@ -122,20 +122,17 @@ def setup(chip):
                  ",".join(['pdk', pdkname, 'aprtech', 'openroad', stackup, libtype, 'lef']),
                  step=step, index=index)
 
-        corners = get_corners(chip)
         for lib in targetlibs:
-            for corner in corners:
-                chip.add('tool', tool, 'task', task, 'require',
-                         ",".join(['library', lib, 'output', corner, delaymodel]),
+            for timing_key in get_library_timing_keypaths(chip, lib).values():
+                chip.add('tool', tool, 'task', task, 'require', ",".join(timing_key),
                          step=step, index=index)
             chip.add('tool', tool, 'task', task, 'require',
                      ",".join(['library', lib, 'output', stackup, 'lef']),
                      step=step, index=index)
         for lib in macrolibs:
-            for corner in corners:
-                if chip.valid('library', lib, 'output', corner, delaymodel):
-                    chip.add('tool', tool, 'task', task, 'require',
-                             ",".join(['library', lib, 'output', corner, delaymodel]),
+            for timing_key in get_library_timing_keypaths(chip, lib).values():
+                if chip.valid(*timing_key):
```

Comment:
Ok I see here is another validity check.
If you remove the possiby invalid keypath from `get_library_timing_keypaths()` I think you could remove this one.
If not then you should probably also check validity for the `targetlibs`.

Reply:
macros don't always have a timing model, but the target libs must have one